src/benchmarks/micro/coreclr/Devirtualization/GuardedThreeClassVirtual.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+// Performance test for virtual call dispatch with three
+// possible target classes mixed in varying proportions.
+
+namespace GuardedDevirtualizationThreeClass
+{
+
+public class B
+{
+    public virtual int F() => 33;
+}
+
+public class D : B
+{
+    public override int F() => 44;
+}
+
+public class E : B
+{
+    public override int F() => 55;
+}
+
+public class TestInput
+{
+    public const int N = 1000;
+
+    public TestInput(double pB, double pD)
+    {
+        _pB = pB;
+        _pD = pD;
+        b = GetArray();
+    }
+
+    static Random r = new Random(42);
+
+    double _pB;
+    double _pD;
+    B[] b;
+
+    public B[] Array => b;
+    public override string ToString() => $"pB={_pB:F2} pD={_pD:F2}";
+
+    B[] GetArray()
+    {
+        B[] result = new B[N];
+
+        for (int i = 0; i < N; i++)
+        {
+            double p = r.NextDouble();
+
+            if (p <= _pB)
+            {
+                result[i] = new B();
+            }
+            else if (p <= _pB + _pD)
+            {
+                result[i] = new D();
+            }
+            else
+            {
+                result[i] = new E();
+            }
+        }
+        return result;
+    }
+}
+
+public class Virtual
+{
+    [BenchmarkCategory(Categories.CoreCLR, Categories.Virtual)]
+    [Benchmark(OperationsPerInvoke=TestInput.N)]
+    [ArgumentsSource(nameof(GetInput))]
+    public static long Call(TestInput testInput)
```

Comment:
BDN requires it to be an instance method.

I just created an isse to give a good warning about that: https://github.com/dotnet/BenchmarkDotNet/issues/983

```suggestion
    public long Call(TestInput testInput)
```

Reply:
Thanks -- I was sure this was working when run normally under BDN. Anyways, easy fix.